cupy/indexing/generate.py

```diff
@@ -1,13 +1,167 @@
 # flake8: NOQA
 # "flake8: NOQA" to suppress warning "H104  File contains nothing but comments"
 
-
-# class c_(object):
-# class r_(object):
 # class s_(object):
 
 import numpy
+
 import cupy
+
+import six
+
+
+class AxisConcatenator(object):
```

Comment:
In NumPy, `RClass` does call `AxisConcatenator.__init__(self, 0)` (see [here](https://github.com/numpy/numpy/blob/master/numpy/lib/index_tricks.py#L454)). But as `0` is same as the default argument, it does not have any effect. The current PR is the result of the removal of this argument.

But as you point out, `AxisConcatenator` and `RClass` are equivalent. If we do not mind the inconsistency about the class hierarchies between NumPy and CuPy and non-existence of `AxisConcatenator`, it is OK to remove `AxisConcatenator` and move its implementation to `RClass`.